src/Services/ProductRegistrationService.php

```diff
@@ -56,14 +56,14 @@ public function getProductStatus(): array
             }
         } else {
             // No change in version, so do not show the dialog if has opted out
-            if ($telemetry_disabled == null || $optOut == null) {
+            if ($telemetry_disabled === null || $optOut == null) {
                 $row['allowRegisterDialog'] = 1;
-                if ($optOut == null) {
-                    $row['allowEmail'] = 1;
-                }
-                if ($telemetry_disabled == null) {
-                    $row['allowTelemetry'] = 1;
-                }
+            }
+            if ($optOut == null) {
+                $row['allowEmail'] = 1;
+            }
+            if (empty($telemetry_disabled)) {
+                $row['allowTelemetry'] = 1;
```

Comment:
This will make allowEmail and allowTelemetry on the dialog no longer completely decoupled. In this case, if, optout for email is null, then the dialog will show for telemetry even if it has been opted out. I'll post a fix for this. Kicking myself that i should of made these variables more clear that only for whether to display in the dialog.

Reply:
I tested this and was working for me. Although I may have missed an update from just being tired. I'll test again but rename at will. I wished I had named telemetry_disable to telemetry enabled. That bit toggle just got me confused.

Reply:
btw telemetry_disabled is decoupled and L-65 is needed or reporting won't work i.e. 0 and or null.

Reply:
What am I missing Brady
![image](https://github.com/user-attachments/assets/788330d7-6e13-48b6-b22f-1f41f9a684f3)


![image](https://github.com/user-attachments/assets/d908d6c0-22c2-4b6e-b68d-48e8a4fac9fd)

Reply:
yep doesn't work if email is null and opt-out = 0 but I don't see why it'd get in that condition because null was the neutral state but maybe you can think of another event to add while there.:)